lib/mobility-core/src/Kernel/External/Maps/Google/MapsClient.hs

```diff
@@ -167,13 +169,23 @@ distanceMatrix ::
   [GoogleMaps.Place] ->
   [GoogleMaps.Place] ->
   Maybe GoogleMaps.Mode ->
+  Bool ->
   m GoogleMaps.DistanceMatrixResp
-distanceMatrix url key origins destinations mode = do
-  callAPI url (distanceMatrixClient origins destinations key mode) "distanceMatrix"
-    >>= checkGoogleMapsError url
-    >>= \resp -> do
-      mapM_ (mapM validateResponseStatus . (.elements)) resp.rows
-      return resp
+distanceMatrix url key origins destinations mode isAvoidTolls =
+  if isAvoidTolls
```

Comment:
Instead of adding `if` check on whole block, just add if check for the parameter